cmd/datamon/cmd/bundle_get.go

```diff
@@ -57,8 +58,24 @@ exits with ENOENT status otherwise.`,
 			return
 		}
 
+		var labels []model.LabelDescriptor
+		if datamonFlags.bundle.WithLabels {
+			labels = getLabels(remoteStores)
+		}
+
 		var buf bytes.Buffer
-		err = bundleDescriptorTemplate.Execute(&buf, bundle.BundleDescriptor)
+		if len(labels) > 0 {
+			data := struct {
+				model.BundleDescriptor
+				Labels string
+			}{
+				BundleDescriptor: bundle.BundleDescriptor,
+			}
+			data.Labels = displayBundleLabels(bundle.BundleDescriptor.ID, labels)
+			err = bundleDescriptorTemplate(datamonFlags.bundle.WithLabels).Execute(&buf, data)
+		} else {
+			err = bundleDescriptorTemplate(datamonFlags.bundle.WithLabels).Execute(&buf, bundle.BundleDescriptor)
```

Comment:
`model.BundleDescriptor` doesn't have a `Labels` `reflect.StructField`, so if `datamonFlags.bundle.WithLabels` is true and `len(labels)` is zero (i.e. if the `--with-labels` flag is passed and the bundle in question happens not to have any labels, this this will pass `true` to `bundleDescriptorTemplate` and we'll be attempting to call this `Execute` with a template string

```
{{.ID}} ,{{.Labels}}, {{.Timestamp}} , {{.Message}}
```

and a `model.BundleDescriptor` `interface{}` parameter, which will result in an error?

Reply:
@ransomw1c ok will do.

Reply:
@ransomw1c there is no failure when no label is set on the bundle. I am now adding a test to prove that.
TL,DR:
```
./datamon bundle get --repo fred-test-repo --with-labels
Using bundle: 1WAKWqimd3kKt24oI3qg3XGzXsu
{"level":"info","ts":1578641986.0438097,"caller":"core/bundle_unpack.go:258","msg":"kicking off filelist download","concurrent Filelist Downloads":10}
{"level":"info","ts":1578641986.044021,"caller":"core/bundle_unpack.go:277","msg":"preallocating bundle entries","max entries":0}
1WAKWqimd3kKt24oI3qg3XGzXsu ,<no label>, 2020-01-09 17:21:54.772393594 +0100 CET , xyz
```

```
./datamon label set --bundle 1WAKWqimd3kKt24oI3qg3XGzXsu  --repo fred-test-repo --label my-label
./datamon bundle get --repo fred-test-repo --with-labels
Using bundle: 1WAKWqimd3kKt24oI3qg3XGzXsu
{"level":"info","ts":1578642033.7185533,"caller":"core/bundle_unpack.go:258","msg":"kicking off filelist download","concurrent Filelist Downloads":10}
{"level":"info","ts":1578642033.7187045,"caller":"core/bundle_unpack.go:277","msg":"preallocating bundle entries","max entries":0}
1WAKWqimd3kKt24oI3qg3XGzXsu ,(my-label), 2020-01-09 17:21:54.772393594 +0100 CET , xyz
```

Reply:
```
./datamon bundle list --repo fred-test-repo --with-labels
1VqLCB9upklFMAr3kch5Xhv2FXv ,<no label>, 2020-01-02 15:32:27.948443111 +0100 CET , upload a bundle
1VqM633GnZ8Mh1Dt38rXWlcDKHD ,<no label>, 2020-01-02 15:39:21.299499846 +0100 CET , Mutable mount
1VqMMh2XqjM9JZVRvsK5BJfcf5b ,<no label>, 2020-01-02 15:41:22.454642726 +0100 CET , Mutable mount
1VqMZtZzCRtoWSQIgarW9EBaoEl ,(mutable-mount-fred), 2020-01-02 15:42:28.202893153 +0100 CET , Mutable mount
1VqMoiNzR2ZXaaoQfzhXj6JXsP7 ,(mutable-mount-fred-4), 2020-01-02 15:45:03.56740456 +0100 CET , Mutable mount 4
1VqNBo70inzbiRTPkleMrl1mhVR ,<no label>, 2020-01-02 15:48:50.890137366 +0100 CET , upload a bundle again
1VqRkcyVagyL8ssEfo1uCI7pB3i ,(mutable-mount-fred-5), 2020-01-02 15:52:37.756544005 +0100 CET , Mutable mount 5
1VqRwfofF0To4d9OY5VgAAxrVxN ,<no label>, 2020-01-02 16:27:57.498375401 +0100 CET , upload a bundle yet again
1VqS4BwCRSIFSFQDAmqc2K4NMdv ,(mutable-mount-fred-6), 2020-01-02 16:28:55.944993424 +0100 CET , Mutable mount 6
1VqSA0iNLRIr6L3TpuWNloFJRov ,(un label), 2020-01-02 16:29:44.555730364 +0100 CET , upload a bundle yet again
1VqSirBWrUDC9Pa0KQIiJp64sUW ,(mutable-mount-fred-7), 2020-01-02 16:30:43.598354541 +0100 CET , Mutable mount 7
1WAKWqimd3kKt24oI3qg3XGzXsu ,(my-label), 2020-01-09 17:21:54.772393594 +0100 CET , xyz
```

Reply:
@ransomw1c corner case of repo without label fixed, CLI test added to cover that case